framework/src/modules/interoperability/utils.ts

```diff
@@ -593,3 +618,347 @@ export const commonCCUExecutelogic = async (args: CommonExecutionLogicArgs) => {
 	}
 	await partnerChannelStore.setWithSchema(chainIDBuffer, partnerChannelData, channelSchema);
 };
+
+export const initGenesisStateUtil = async (id: Buffer, ctx: GenesisBlockExecuteContext) => {
+	const assetBytes = ctx.assets.getAsset(id);
+	if (!assetBytes) {
+		return;
+	}
+
+	const { getStore } = ctx;
+
+	const genesisStore = codec.decode<GenesisInteroperabilityStore>(
+		genesisInteroperabilityStoreSchema,
+		assetBytes,
+	);
+	const errors = validator.validate(genesisInteroperabilityStoreSchema, genesisStore);
+	if (errors.length) {
+		throw new LiskValidationError(errors);
+	}
+
+	const outboxRootStoreKeySet = new dataStructures.BufferSet();
+	const outboxRootStore = getStore(id, STORE_PREFIX_OUTBOX_ROOT);
+	for (const outboxRootData of genesisStore.outboxRootSubstore) {
+		if (outboxRootStoreKeySet.has(outboxRootData.storeKey)) {
+			throw new Error(
+				`Outbox root store key ${outboxRootData.storeKey.toString('hex')} is duplicated.`,
+			);
+		}
+		outboxRootStoreKeySet.add(outboxRootData.storeKey);
+		await outboxRootStore.setWithSchema(
+			outboxRootData.storeKey,
+			outboxRootData.storeValue,
+			outboxRootSchema,
+		);
+	}
+
+	const ownChainAccountStore = getStore(id, STORE_PREFIX_OWN_CHAIN_DATA);
+	const ownChainAccount = await ownChainAccountStore.getWithSchema<OwnChainAccount>(
+		MAINCHAIN_ID_BUFFER,
+		ownChainAccountSchema,
+	);
+	const channelDataStoreKeySet = new dataStructures.BufferSet();
+	const channelDataStore = getStore(id, STORE_PREFIX_CHANNEL_DATA);
+	for (const channelData of genesisStore.channelDataSubstore) {
+		if (channelDataStoreKeySet.has(channelData.storeKey)) {
+			throw new Error(
+				`Channel data store key ${channelData.storeKey.toString('hex')} is duplicated.`,
+			);
+		}
+		channelDataStoreKeySet.add(channelData.storeKey);
+
+		const tokenID = channelData.storeValue.messageFeeTokenID;
+		if (
+			!(
+				tokenID.chainID.readInt32BE(0) === 1 ||
+				tokenID.chainID.equals(channelData.storeKey) ||
+				tokenID.chainID.equals(ownChainAccount.id)
+			)
+		) {
+			throw new Error(
+				`Chain id corresponding to the channel data store key ${channelData.storeKey.toString(
+					'hex',
+				)} is not valid.`,
+			);
+		}
+		if (tokenID.chainID.equals(MAINCHAIN_ID_BUFFER) && tokenID.localID.readInt32BE(0) !== 0) {
+			throw new Error(
+				`Local id corresponding to the channel data store key ${channelData.storeKey.toString(
+					'hex',
+				)} is not valid.`,
+			);
+		}
```

Comment:
Either channel.messageFeeTokenID={chainID=1, localID=0} (corresponding to the LSK token), or channel.messageFeeTokenID.chainID must be equal to channelData.storeKey or ownChainAccount.ID (the message fee token must be a native token of either chains).